### PR TITLE
Implement protocol v2 decoding and reading

### PR DESCRIPTION
**Pull request by oktomus**
Needs to be merged along with [ Allow sending AOV tiles to stdout #1906 ](https://github.com/appleseedhq/appleseed/pull/1906)

### CODE REVIEW

File: render.py [hunk 1]
```diff
@@ -235,14 +235,27 @@ def __render_project_file(self, scene, project_filepath, project_dir=None):
...
 
+            # Protocol V1
             if chunk_type == 1:
-                # Tile data (protocol v1).
+                # Tile data, protocol v1
                 if not self.__process_tile_data_chunk(process, min_x, min_y, max_x, max_y):
                     break
-            elif chunk_type == 2:
-                # Tile highlight (protocol v1).
+            elif (chunk_type == 2) or (chunk_type == 10):
```
Comment (dictoon) [inline render.py]: Remove redundant parentheses here and in other relevant places.

File: render.py [hunk 2]
```diff
@@ -387,6 +402,119 @@ def __process_tile_highlight_chunk(self, process, min_x, min_y, max_x, max_y):
...
+
+        aov_name = os.read(process.stdout.fileno(), aov_name_len)
+
+        return True
+
+    def __process_plane_data_chunk(self, process, min_x, min_y, max_x, max_y):
+        # Read and decode plane header and content.
+        # Protocol v2
+        # Contains:
+        #  Aov index
```
Comment (dictoon) [inline render.py]: *AOV*

File: render.py [hunk 3]
```diff
@@ -387,6 +402,119 @@ def __process_tile_highlight_chunk(self, process, min_x, min_y, max_x, max_y):
 
         return True
 
+    def __process_tiles_header(self, process):
+        # Read and decode tiles header
```
Comment (dictoon) [inline render.py]: In Python it's actually better to use docstrings: https://www.python.org/dev/peps/pep-0257/

File: render.py [hunk 4]
```diff
@@ -387,6 +402,119 @@ def __process_tile_highlight_chunk(self, process, min_x, min_y, max_x, max_y):
...
+        #  Tile pixels
+        tile_header = struct.unpack("IIIIII", os.read(process.stdout.fileno(), 6 * 4))
+        tile_aov_index = tile_header[0]
+        tile_x = tile_header[1]
+        tile_y = tile_header[2]
+        tile_w = tile_header[3]
+        tile_h = tile_header[4]
+        tile_c = tile_header[5]
+
+        # We process only the beauty AOV
```
Comment (dictoon) [inline render.py]: End sentence with a point.

**Comment by oktomus** [comment]
@dictoon Yes, this one can be merged before the other, but not the opposite.

**Pull request merged**